CMakeLists.txt

```diff
@@ -185,13 +185,37 @@ if(${CMAKE_VERSION} VERSION_GREATER 3.7.99)
   set(USING_CUDA_LANG_SUPPORT True)
   set(CMAKE_CUDA_STANDARD 11)
   set(CMAKE_CUDA_STANDARD_REQUIRED True)
+  execute_process(
+    COMMAND "${CUDA_HOST_COMPILER}"
+    -V
+    OUTPUT_VARIABLE output ERROR_VARIABLE output
+    RESULT_VARIABLE result
+    TIMEOUT 10
+    )
+  if (output MATCHES "Intel")
+    message(FATAL_ERROR "QUDA's CUDA_HOST_COMPILER must not be Intel builds. Please set CUDA_HOST_COMPILER to a compatible CXX compiler (g++,clang++).")
+  endif()
 else()
-  set(CUDA_HOST_COMPILER "${CMAKE_CXX_COMPILER}" CACHE FILEPATH "Host side compiler used by NVCC")
   mark_as_advanced(CUDA_HOST_COMPILER)
   find_package(CUDA REQUIRED)
   set(USING_CUDA_LANG_SUPPORT False)
+  execute_process(
+    COMMAND "${CUDA_HOST_COMPILER}"
+    -V
+    OUTPUT_VARIABLE output ERROR_VARIABLE output
+    RESULT_VARIABLE result
+    TIMEOUT 10
+    )
+  if (output MATCHES "Intel")
+    message(FATAL_ERROR "QUDA's CUDA_HOST_COMPILER must not be Intel builds. Please set CUDA_HOST_COMPILER to a compatible CXX compiler (g++,clang++).")
```

Comment:
guess this is ok for now. Wondering whether we can get rid of this code and exploit cmake's compiler detection.